exercises/change/example.js

```diff
@@ -0,0 +1,139 @@
+class Change {}
+
+export default Change;
+
+// data structure to hold each candidate solution that is generated
+class Candidate {
+  constructor() {
+    this.wasSearched = false;
+    this.coins = [];
+  }
+
+  searched() {
+    this.wasSearched = true;
+  }
+
+  isSearched() {
+    return this.wasSearched;
+  }
+
+  getCoins() {
+    return this.coins;
+  }
+
+  addCoin(coin) {
+    const sortNum = (a, b) => a - b;
+
+    this.coins.push(coin);
+    this.coins.sort(sortNum);
+  }
+
+  getCoinCount() {
+    return this.coins.length;
+  }
+
+  getSum() {
+    const getSum = (total, num) => total + num;
+    return this.coins.reduce(getSum);
+  }
+}
+
+Change.prototype.calculate = (coinArray, target) => {
+  const candidates = [];
+  // fill the array with 0 to start
+  candidates[target] = 0;
+  candidates.fill(0);
+
+  // save a new candidate to the candidate array
+  const saveCandidate = (candidate) => {
+    const sum = candidate.getSum();
+
+    if (candidate.getSum() <= target &&
+            typeof (candidates[sum]) !== 'number' &&
```

Comment:
I'm wondering if this wouldn't be more readable if the conditional were extracted into a named function instead?

Also, the indenting is a little unusual here.  You could run the linter against your code in "fix mode" to perhaps take care of a lot of these things automatically.  Let me know if I can provide some assistance on how to do that.